webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultCsvValidationService.java

```diff
@@ -121,68 +134,91 @@ public DefaultCsvValidationService(final SchoolRepository schoolRepository) {
                                       final String school,
                                       final String group,
                                       final int studentCount,
                                       final long recordNumber) {
         if (studentCount < MaxStudentsPerGroup) return;
 
-        validationFailures.add(CsvValidationFailure.builder()
+        validationFailures.add(CsvValidationResult.builder()
                 .row(recordNumber)
                 .message("School [" + school + "] Group [" + group + "] contains " + studentCount + " students.  Maximum students allowed per group is " + MaxStudentsPerGroup)
+                .isOk(false)
                 .build());
     }
 
-    private void validateSchoolGroupYear(final List<CsvValidationFailure> validationFailures,
+    private void validateSchoolGroupYear(final List<CsvValidationResult> validationFailures,
                                          final Set<String> completedSchoolGroupYears,
                                          final String schoolGroupYear,
                                          final long recordNumber) {
         if (completedSchoolGroupYears.contains(schoolGroupYear)) {
-            validationFailures.add(CsvValidationFailure.builder()
+            validationFailures.add(CsvValidationResult.builder()
                     .row(recordNumber)
                     .message("Record is out-of-order: sort CSV by school, group, year")
+                    .isOk(false)
                     .build());
         }
     }
 
     @Override
-    public String toMessage(final List<CsvValidationFailure> failures) {
+    public String toFailureMessage(final List<CsvValidationResult> failures) {
         final StringBuilder messageBuilder = new StringBuilder();
-        final List<CsvValidationFailure> subList = failures.size() < MaxFailureMessages ? failures : failures.subList(0, MaxFailureMessages);
-        for (final CsvValidationFailure failure : subList) {
-            messageBuilder.append(String.format(ValidationFormat, failure.getRow(), failure.getMessage()));
+        final List<CsvValidationResult> subList = failures.size() < MaxMessages ? failures : failures.subList(0, MaxMessages);
+        for (final CsvValidationResult validationResult : subList) {
+            messageBuilder.append(String.format(ValidationFailureFormat, validationResult.getRow(), validationResult.getMessage()));
         }
-        if (failures.size() > MaxFailureMessages) {
-            messageBuilder.append(String.format(MoreFailuresFormat, failures.size() - MaxFailureMessages));
+        if (failures.size() > MaxMessages) {
+            messageBuilder.append(String.format(MoreFailuresFormat, failures.size() - MaxMessages));
+        }
+        return messageBuilder.toString();
+    }
+
+    @Override
+    public String toSuccessMessage(final List<CsvValidationResult> successes) {
+        final StringBuilder messageBuilder = new StringBuilder();
```

Comment:
I think a StringJoiner would make this slightly more readable.

Reply:
If java string builders had an `appendLine` it would also be slightly more readable :) 

I can use a StringJoiner instead though.